src/components/ThemeModal/index.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect, useState } from 'react';
+import styled, { useTheme } from 'styled-components';
+
+// Components
+import { BackButton, HeaderTitle, ModalHeader } from '../Menu/MenuModalWrapper';
+
+// Hooks
+import { useEtherspot } from '../../hooks';
+
+// Icons
+import { IoMdCheckmark } from 'react-icons/io';
+import { CloseButton } from '../Button';
+
+// Utils
+import { THEME_TYPES, Theme, getTheme } from '../../utils/theme';
+
+const ThemeModal = ({ onBackButtonClick }: { onBackButtonClick: () => void }) => {
+  const [appliedTheme, setAppliedTheme] = useState<string>('');
```

Comment:
Nice thing about enums is that we can use them as types, also good to have a default
```suggestion
  const [appliedTheme, setAppliedTheme] = useState<ThemeType>(ThemeType.DARK);
```